### PR TITLE
Dennis en passant code

### PR DESCRIPTION
**Pull request by dennisgit42**
I spent all day building on what Melsen had, but can't tell why the app is still telling me it is not a valid move when I try to drag and drop a piece to do en passant capture.

### CODE REVIEW

**Comment by MelsenC** [comment]
Dennis, there was a small mistake in my code. I have since fixed it and written outlines for tests... 

Apologies- I'm going to the [counter-protest](https://www.facebook.com/events/165697793990647) march today on Boston Common, and I don't know how long it will go. Will come back to this when I get back this afternoon.

**Comment by MelsenC** [comment]
(sorry - didn't mean to close this pull request!)

**Comment by dennisgit42** [comment]
Still trying to get jquery drag and drop to automatically remove the pawn captured through en passant without a page refresh, but the database change appears to work

**Comment by dennisgit42** [comment]
This should be ready to merge into master, drag and drop now works with en passant

File: app/models/pawn.rb [hunk 1]
```diff
@@ -1,49 +1,74 @@
...
+    return true if capture_move?(new_x.to_i, new_y.to_i)
+    return true if en_passant_capture?(new_x.to_i, new_y.to_i)
+    if allowed_to_move?(new_x.to_i, new_y.to_i) && !square_occupied?(new_x.to_i, new_y.to_i)
+      update_attributes(turn_pawn_moved_twice: game.move_number + 1) if moving_two_squares?(new_x.to_i, new_y.to_i)
+      return true
+    end
+    false
+  end
+
+  def vul_to_en_passant?
```
Comment (mscoutermarsh) [inline app/models/pawn.rb]: what's "vul" mean?

File: app/models/pawn.rb [hunk 2]
```diff
@@ -1,49 +1,74 @@
...
+    game.move_number == turn_pawn_moved_twice && piece_move_number == 1 && (y_position == 4 || y_position == 5)
+  end
+
+  def en_passant_capture?(new_x, new_y)
```
Comment (mscoutermarsh) [inline app/models/pawn.rb]: Can this method be private?

File: app/models/piece.rb [hunk 3]
```diff
@@ -58,6 +60,13 @@ def obstructed?(new_x, new_y)
 
   private
 
+  def increment_move
+    game.update_attributes(move_number: game.move_number + 1)
+    update_attributes(game_move_number: game.move_number)
+    update_attributes(piece_move_number: piece_move_number + 1)
+    update_attributes(has_moved: true)
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: Could combine 3 of these lines into 1 to reduce database queries.

`update_attributes(game_move_number: game.move_number, piece_move_number:...)`

Each call to `update_attributes` = 1 query

**Pull request merged**